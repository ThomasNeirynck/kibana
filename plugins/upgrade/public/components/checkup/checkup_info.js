import PropTypes from 'prop-types';
import React from 'react';

import { Issues } from './issues';


const INFO_ISSUES = [
  {
    level: 'none',
    message: 'Everything is OK.',
  },
  {
    level: 'info',
    message: 'Info that something has changed. No action required to upgrade.',
  },
  {
    level: 'warning',
    message: (
      'Action recommended but not required to upgrade. You are using '
      + 'deprecated functionality which will not be available in version 6.0.'
    ),
  },
  {
    level: 'critical',
    message: 'Action required. You must fix this problem to upgrade.',
  },
];

export function CheckupInfo({ className }) {
  return (
    <div className={className}>
      <p className="kuiText kuiVerticalRhythm">
        This tool runs a series of checks against your Elasticsearch cluster, nodes, and indices
        to determine whether you can upgrade directly to Elasticsearch version 6, or whether you
        need to make changes to your data before doing so.
      </p>
      <p className="kuiText kuiVerticalRhythm">
        You will also see deprecated cluster settings and node settings, deprecated plugins, and
        indices with deprecated mappings currently in use.
      </p>
      <p className="kuiText kuiVerticalRhythm">
        Each issue found has a degree of severity:
      </p>
      <Issues className="kuiVerticalRhythm" issues={INFO_ISSUES} />
    </div>
  );
}

CheckupInfo.propTypes = {
  className: PropTypes.string,
};

CheckupInfo.defaultProps = {
  className: null,
};
