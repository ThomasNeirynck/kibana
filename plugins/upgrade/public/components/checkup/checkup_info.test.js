import { shallow } from 'enzyme';
import React from 'react';

import { CheckupInfo } from './checkup_info';

jest.mock('ui/chrome', () => {}, { virtual: true });
jest.mock('ui/notify/notifier', () => ({
  Notifier: function () {},
}), { virtual: true });


describe('CheckupInfo', () => {
  test('renders the className property', () => {
    const component = (
      <CheckupInfo className="myClass" />
    );

    expect(shallow(component)).toMatchSnapshot();
  });
});
