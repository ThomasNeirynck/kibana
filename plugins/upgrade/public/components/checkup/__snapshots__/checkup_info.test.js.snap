// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckupInfo renders the className property 1`] = `
<div
  className="myClass"
>
  <p
    className="kuiText kuiVerticalRhythm"
  >
    This tool runs a series of checks against your Elasticsearch cluster, nodes, and indices to determine whether you can upgrade directly to Elasticsearch version 6, or whether you need to make changes to your data before doing so.
  </p>
  <p
    className="kuiText kuiVerticalRhythm"
  >
    You will also see deprecated cluster settings and node settings, deprecated plugins, and indices with deprecated mappings currently in use.
  </p>
  <p
    className="kuiText kuiVerticalRhythm"
  >
    Each issue found has a degree of severity:
  </p>
  <Issues
    className="kuiVerticalRhythm"
    issues={
      Array [
        Object {
          "level": "none",
          "message": "Everything is OK.",
        },
        Object {
          "level": "info",
          "message": "Info that something has changed. No action required to upgrade.",
        },
        Object {
          "level": "warning",
          "message": "Action recommended but not required to upgrade. You are using deprecated functionality which will not be available in version 6.0.",
        },
        Object {
          "level": "critical",
          "message": "Action required. You must fix this problem to upgrade.",
        },
      ]
    }
  />
</div>
`;
